### PR TITLE
[Merged by Bors] - chore(Algebra/Lie/Submodule): the lattice of Lie submodules is atomic

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/Submodule.lean [hunk 1]
```diff
@@ -1545,3 +1549,29 @@ def LieModuleEquiv.ofTop : (⊤ : LieSubmodule R L M) ≃ₗ⁅R,L⁆ M :=
...
+termination_by N => finrank K N
+decreasing_by exact finrank_lt_finrank_of_lt hbN
+
+instance [FiniteDimensional K V] : IsAtomic (LieSubmodule K L V) := by
+  constructor
+  intro N
+  rw [or_iff_not_imp_left]
+  exact exists_atom_le_of_finite N
+
+end LieSubmodule
```
Comment (ocfnash) [inline Mathlib/Algebra/Lie/Submodule.lean:1577]: I think you can drop all of this and just use:
```lean
instance [IsArtinian R M] : IsAtomic (LieSubmodule R L M) :=
  isAtomic_of_orderBot_wellFounded_lt <| wellFounded_of_isArtinian R L M
```
beside `wellFounded_of_isArtinian`.

**Review by ocfnash — COMMENTED** [review]
I think

**Comment by jcommelin** [comment]
* **#13387** <a href="https://app.graphite.dev/github/pr/leanprover-community/mathlib4/13387?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jcommelin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: Mathlib/Algebra/Lie/Submodule.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/Algebra/Lie/Submodule.lean]: Great advice! Done

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**